src/Dialogs/StatusPanels/TimesStatusPanel.cpp

```diff
@@ -97,7 +97,14 @@ TimesStatusPanel::Refresh() noexcept
   }
 
   if (flight.flight_time.count() > 0) {
-    SetText(FlightTime, FormatSignedTimeHHMM(flight.flight_time));
+    if (flight.takeoff_time.IsDefined() && flight.landing_time.IsDefined()) {
+      RoughTime rough_takeoff_time = RoughTime::FromSinceMidnight(flight.takeoff_time);
+      RoughTime rough_landing_time = RoughTime::FromSinceMidnight(flight.landing_time);
+      SetText(FlightTime, FormatSignedTimeHHMM(FloatDuration(rough_landing_time-rough_takeoff_time)));
```

Comment:
Is this robust against midnight rollover?

Reply:
It's not robust against midnight rollover. One mistake is your new `FromSinceMidnight()` implementation, and in general your use of `class RoughTime`. That class cannot have values beyond 24 hours. Therefore, you cannot use it to calculate time stamps which may have exceeded to the next day (which is very common in certain time zones).
Your code is not only complicated and slow, but also wrong.

Reply:
I was uncertain myself if that will be 24h rollover safe and tested it. The outcome with the igc replay was positive. Starting one day landing the next day is handled correctly. I am pretty sure that for flights beyond the second 24h rollover we will miss 24h on flight time. 
Generally I was also not happy to see this limitation in the time stamp implementation serving as a time base for xcs. But that is worth a discussion. Will kick this off.
Thank you, Max, for reviewing carefully.